src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java

```diff
@@ -239,9 +255,9 @@ else if (b instanceof BuildingBarracksTower)
     /**
      * Spawn a brand new Citizen.
      */
-    public void spawnCitizen(@NotNull final Colony colony)
+    public void spawnCitizen()
     {
-        spawnCitizen(null, colony.getWorld(), colony.getBuildingManager(), colony);
+        spawnCitizen(null, colony.getWorld());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Parameter 1 to this call is marked "org.jetbrains.annotations.NotNull" but null is passed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2637)